### PR TITLE
Automatically derive titles from md docs

### PR DESCRIPTION
**Pull request by ras0219-msft**
This PR derives nav titles based on the first line of the markdown doc under translation, assuming it matches `^#* (.*)$`.

### CODE REVIEW

File: en/docs/README.html [hunk 1]
```diff
@@ -71,18 +71,18 @@
       </div>
 <ul class=docs-navigation><li class="list-can-expand"><button class="button-list-can-expand">about</button></li>
 <ul class="collapse standard-padding">
-<a class="doc-outline-link" href="/en/docs/about/faq.html"><li class="list-can-expand">faq</li></a>
-<a class="doc-outline-link" href="/en/docs/about/privacy.html"><li class="list-can-expand">privacy</li></a>
+<a class="doc-outline-link" href="/en/docs/about/faq.html"><li class="list-can-expand">Frequently Asked Questions</li></a>
+<a class="doc-outline-link" href="/en/docs/about/privacy.html"><li class="list-can-expand">Vcpkg telemetry and privacy</li></a>
```
Comment (BillyONeal) [inline en/docs/README.html:75]:
```suggestion
<a class="doc-outline-link" href="/en/docs/about/privacy.html"><li class="list-can-expand">vcpkg Telemetry and Privacy</li></a>
```

File: en/docs/README.html [hunk 2]
```diff
@@ -71,18 +71,18 @@
...
-<a class="doc-outline-link" href="/en/docs/examples/packaging-zipfiles.html"><li class="list-can-expand">Packaging Zipfiles</li></a>
-<a class="doc-outline-link" href="/en/docs/examples/patching.html"><li class="list-can-expand">patching</li></a>
+<a class="doc-outline-link" href="/en/docs/examples/installing-and-using-packages.html"><li class="list-can-expand">Installing and Using Packages Example: SQLite</li></a>
+<a class="doc-outline-link" href="/en/docs/examples/manifest-mode-cmake.html"><li class="list-can-expand">Manifest Mode: CMake Example</li></a>
+<a class="doc-outline-link" href="/en/docs/examples/modify-baseline-to-pin-old-boost.html"><li class="list-can-expand">Pin old Boost versions</li></a>
+<a class="doc-outline-link" href="/en/docs/examples/overlay-triplets-linux-dynamic.html"><li class="list-can-expand">Overlay triplets example</li></a>
+<a class="doc-outline-link" href="/en/docs/examples/packaging-github-repos.html"><li class="list-can-expand">Packaging Github Repos Example: libogg</li></a>
+<a class="doc-outline-link" href="/en/docs/examples/packaging-zipfiles.html"><li class="list-can-expand">Packaging Zipfiles Example: zlib</li></a>
```
Comment (BillyONeal) [inline en/docs/README.html:84]:
```suggestion
<a class="doc-outline-link" href="/en/docs/examples/packaging-zipfiles.html"><li class="list-can-expand">Packaging .zip Files Example: zlib</li></a>
```

File: en/docs/README.html [hunk 3]
```diff
@@ -71,18 +71,18 @@
...
+<a class="doc-outline-link" href="/en/docs/examples/patching.html"><li class="list-can-expand">Patching Example: Patching libpng to work for x64-uwp</li></a>
```
Comment (BillyONeal) [inline en/docs/README.html:85]:
```suggestion
<a class="doc-outline-link" href="/en/docs/examples/patching.html"><li class="list-can-expand">Patching Example: Patching libpng for x64-uwp</li></a>
```

File: en/docs/README.html [hunk 4]
```diff
@@ -178,34 +178,34 @@
...
-<a class="doc-outline-link" href="/en/docs/users/selecting-library-features.html"><li class="list-can-expand">Selecting Library Features</li></a>
-<a class="doc-outline-link" href="/en/docs/users/triplets.html"><li class="list-can-expand">triplets</li></a>
-<a class="doc-outline-link" href="/en/docs/users/versioning.implementation-details.html"><li class="list-can-expand">Versioning.implementation Details</li></a>
-<a class="doc-outline-link" href="/en/docs/users/versioning.html"><li class="list-can-expand">versioning</li></a>
+<a class="doc-outline-link" href="/en/docs/users/integration.html"><li class="list-can-expand">Buildsystem Integration</li></a>
+<a class="doc-outline-link" href="/en/docs/users/manifests.html"><li class="list-can-expand">Manifest Mode</li></a>
+<a class="doc-outline-link" href="/en/docs/users/mingw.html"><li class="list-can-expand">Vcpkg and Mingw-w64</li></a>
+<a class="doc-outline-link" href="/en/docs/users/registries.html"><li class="list-can-expand">Using Registries</li></a>
+<a class="doc-outline-link" href="/en/docs/users/selecting-library-features.html"><li class="list-can-expand">Selecting library features</li></a>
```
Comment (BillyONeal) [inline en/docs/README.html:203]:
```suggestion
<a class="doc-outline-link" href="/en/docs/users/selecting-library-features.html"><li class="list-can-expand">Selecting Library Features</li></a>
```

File: en/docs/README.html [hunk 5]
```diff
@@ -178,34 +178,34 @@
...
+<a class="doc-outline-link" href="/en/docs/users/triplets.html"><li class="list-can-expand">Triplet files</li></a>
```
Comment (BillyONeal) [inline en/docs/README.html:204]:
```suggestion
<a class="doc-outline-link" href="/en/docs/users/triplets.html"><li class="list-can-expand">Triplet Files</li></a>
```

**Review by BillyONeal — COMMENTED** [review]
I think some of the casing is wrong; do I read from the description that that should be fixed in the markdown?

**Comment by ras0219-msft** [comment]
That is the intent of my PR, yes, that the markdown documents can serve as the source of truth for those kinds of things. When we add new doc files, we should be able to simply and directly regenerate the html with no tweaking on this side.

**Comment by BillyONeal** [comment]
Can you apply https://github.com/microsoft/vcpkg/pull/19345 and then rerun your script?

**Review by BillyONeal — APPROVED** [review]
Please rerun your script to pick up changes from microsoft/vcpkg#19345

**Review by strega-nil-ms — APPROVED** [review]

**Pull request merged**